presto-orc/src/test/java/com/facebook/presto/orc/writer/TestColumnWriters.java

```diff
@@ -0,0 +1,144 @@
+    public Object[][] dataForSequenceIdTest()
+    {
+        Block stringBlock = VARCHAR.createBlockBuilder(null, 2)
+                .appendNull()
+                .writeBytes(Slices.utf8Slice("123456789"), 0, 9)
+                .closeEntry()
+                .build();
+
+        Type mapType = mapType(INTEGER, INTEGER);
+        BlockBuilder mapBlockBuilder = mapType.createBlockBuilder(null, 3);
+        mapBlockBuilder.appendNull();
+        mapBlockBuilder.beginBlockEntry().writeInt(1).closeEntry().writeInt(2).closeEntry();
+        mapBlockBuilder.closeEntry();
+        Block mapBlock = mapBlockBuilder.build();
+
+        Type arrayType = arrayType(INTEGER);
+        BlockBuilder arrayBlockBuilder = arrayType.createBlockBuilder(null, 2);
+        arrayBlockBuilder.appendNull();
+        arrayBlockBuilder.beginBlockEntry().writeInt(1).writeInt(2);
+        arrayBlockBuilder.closeEntry();
+        arrayBlockBuilder.beginBlockEntry().appendNull();
+        arrayBlockBuilder.closeEntry();
+        Block arrayBlock = arrayBlockBuilder.build();
+
+        Type rowType = rowType(INTEGER);
+        BlockBuilder rowBlockBuilder = rowType.createBlockBuilder(null, 2);
+        rowBlockBuilder.appendNull();
+        rowBlockBuilder.beginBlockEntry().writeInt(1).closeEntry();
+        rowBlockBuilder.closeEntry();
+        rowBlockBuilder.beginBlockEntry().appendNull();
+        rowBlockBuilder.closeEntry();
+        Block rowBlock = rowBlockBuilder.build();
+
+        return new Object[][] {
+                {toOrcTypes(BOOLEAN), BOOLEAN, BOOLEAN.createFixedSizeBlockBuilder(2).appendNull().writeByte(1).build()},
+                {toOrcTypes(TINYINT), TINYINT, TINYINT.createFixedSizeBlockBuilder(2).appendNull().writeByte(1).build()},
+                {toOrcTypes(SMALLINT), SMALLINT, SMALLINT.createFixedSizeBlockBuilder(2).appendNull().writeShort(1).build()},
+                {toOrcTypes(INTEGER), INTEGER, INTEGER.createFixedSizeBlockBuilder(2).appendNull().writeInt(1).build()},
+                {toOrcTypes(BIGINT), BIGINT, BIGINT.createFixedSizeBlockBuilder(2).appendNull().writeLong(1).build()},
+                {toOrcTypes(DOUBLE), DOUBLE, DOUBLE.createFixedSizeBlockBuilder(2).appendNull().writeLong(1).build()},
+                {toOrcTypes(REAL), REAL, REAL.createFixedSizeBlockBuilder(2).appendNull().writeInt(1).build()},
+                {toOrcTypes(TIMESTAMP), TIMESTAMP, TIMESTAMP.createFixedSizeBlockBuilder(2).appendNull().writeLong(1).build()},
+                {toOrcTypes(VARCHAR), VARCHAR, stringBlock},
+                {toOrcTypes(VARBINARY), VARBINARY, stringBlock},
+                {toOrcTypes(arrayType), arrayType, arrayBlock},
+                {toOrcTypes(mapType), mapType, mapBlock},
+                {toOrcTypes(rowType), rowType, rowBlock},
+        };
+    }
+
+    @Test(dataProvider = "dataForSequenceIdTest")
+    public void testSequenceIdPassedAllColumnWriters(List<OrcType> orcTypes, Type type, Block block)
```

Comment:
This test makes sure that all streams, including nested, has the right sequence id.